pkg/gardenlet/controller/shoot/shoot_control_prepare_migration.go

```diff
@@ -0,0 +1,313 @@
+			c.recorder.Event(shoot, corev1.EventTypeWarning, gardencorev1beta1.EventDeleteError, lastErr.Description)
+			_, updateErr := c.updateShootStatusOperationError(shoot, lastErr.Description, *lastErr)
+			return reconcile.Result{}, utilerrors.WithSuppressed(errors.New(lastErr.Description), updateErr)
+		}
+	}
+
+	if err := o.SwitchBackupEntryToTargetSeed(context.TODO()); err != nil {
+		lastErr := gardencorev1beta1helper.LastError(fmt.Sprintf("Could not delete BackupEntry resource in seed: %s", err))
+		c.recorder.Event(shoot, corev1.EventTypeWarning, gardencorev1beta1.EventDeleteError, lastErr.Description)
+		_, updateErr := c.updateShootStatusOperationError(shoot, lastErr.Description, *lastErr)
+		return reconcile.Result{}, utilerrors.WithSuppressed(errors.New(lastErr.Description), updateErr)
+	}
+
+	c.recorder.Event(shoot, corev1.EventTypeNormal, gardencorev1beta1.EventMigrationPrepared, "Shoot Control Plane prepared for migration, successfully")
+	newShoot, err := kutil.TryUpdateShootAnnotations(c.k8sGardenClient.GardenCore(), retry.DefaultRetry, o.Shoot.Info.ObjectMeta,
+		func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
+			controllerutils.RemoveAllTasks(shoot.Annotations)
+			return shoot, nil
+		},
+	)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	o.Shoot.Info = newShoot
+	_, err = kutil.TryUpdateShootStatus(c.k8sGardenClient.GardenCore(), retry.DefaultRetry, o.Shoot.Info.ObjectMeta,
+		c.updateShootRestorePendingStatusFunc(o.Shoot.Info, "Shoot cluster state has been successfully prepared for migration."))
+	return reconcile.Result{}, err
+}
+
+func (c *Controller) updateShootRestorePendingStatusFunc(shoot *gardencorev1beta1.Shoot, successDescription string) func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
+	return func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
+		shoot.Status.RetryCycleStartTime = nil
+		shoot.Status.SeedName = nil
+		shoot.Status.LastErrors = nil
+		shoot.Status.LastOperation = &gardencorev1beta1.LastOperation{
+			Type:           gardencorev1beta1.LastOperationTypeRestore,
+			State:          gardencorev1beta1.LastOperationStatePending,
+			Progress:       0,
+			Description:    successDescription,
+			LastUpdateTime: metav1.Now(),
+		}
+		return shoot, nil
+	}
+}
+
+func shouldRenewRetryCycleStartTime(o *operation.Operation) bool {
+	return o.Shoot.Info.Status.RetryCycleStartTime == nil ||
+		o.Shoot.Info.Generation != o.Shoot.Info.Status.ObservedGeneration ||
+		o.Shoot.Info.Status.Gardener.Version == version.Get().GitVersion ||
+		o.Shoot.Info.Status.LastOperation != nil && o.Shoot.Info.Status.LastOperation.State == gardencorev1beta1.LastOperationStateFailed
+}
+
+func isShootHibernated(shootHibernation *gardencorev1beta1.Hibernation) bool {
+	return shootHibernation != nil && shootHibernation.Enabled != nil && *shootHibernation.Enabled
+}
```

Comment:
These functions seems duplicative (for the shoot hibernation there is a more accurate helper, and the `shouldRenewRetryCycleStartTime` is never called anyways, so remove it.